### PR TITLE
watchRTC as a feature for Jitsi meet

### PR DESCRIPTION
**Pull request by subhamcyara**
Following PR contains changes required to add watchRTC as a feature for jitsi meet.

This involves adding 
- watchRTC SDK package (npm i @testrtc/watchrtc-sdk).
- configuration to enable watchRTC which includes a **watchRTCEnabled** flag.
- **watchRTCConfigParams** for initializing the SDK.

### CODE REVIEW

**Comment by saghul** [comment]
Can you PTAL @andrei-gavrilescu ?

File: react/features/watchrtc/watchRTCHandler.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/* eslint-enable lines-around-comment */
+
+import { IWatchRTCConfiguration } from '../base/config/configType';
+import logger from './logger';
+
+/**
+ * Class that controls the watchRTC flow, because it overwrites and proxies global function it should only be
+ * initialized once.
+ */
+class watchRTCHandler {
```
Comment (andrei-gavrilescu) [inline react/features/watchrtc/watchRTCHandler.ts:14]: nit. use PascalCase for class name. Also I think there are a couple of linting errors around, consider running the linter.

**Comment by andrei-gavrilescu** [comment]
Looks good to me. One thing to consider is that lib-jitsi-meet users won't have this functionality, that's why we're currently moving rtcstats to lib-jitsi-meet as well see: https://github.com/jitsi/lib-jitsi-meet/pull/2305 and https://github.com/jitsi/jitsi-meet/pull/13508

File: react/features/watchrtc/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+MiddlewareRegistry.register((store: IStore) => (next: Function) => (action: AnyAction) => {
+    const { getState } = store;
+    const state = getState();
+    const config = state['features/base/config'];
+    const { watchRTCConfigParams } = config;
+
+    switch (action.type) {
+    case LIB_WILL_INIT: {
+        if (isRtcstatsEnabled(state)) {
+            logger.error("Cannot initialize WatchRTC when RTCStats is enabled.");
```
Comment (damencho) [inline react/features/watchrtc/middleware.ts:31]: This will be printed for all users using rtcstats, isn't it?
If this is the case, please drop it.

**Comment by saghul** [comment]
> Looks good to me. One thing to consider is that lib-jitsi-meet users won't have this functionality, that's why we're currently moving rtcstats to lib-jitsi-meet as well see:  https://github.com/jitsi/lib-jitsi-meet/pull/2305 and https://github.com/jitsi/jitsi-meet/pull/13508

Agreed. It should also be less code I think.

@subhamcyara can you pl move it to LJM, following the lead of rtcstats?

**Comment by subhamcyara** [comment]
> > Looks good to me. One thing to consider is that lib-jitsi-meet users won't have this functionality, that's why we're currently moving rtcstats to lib-jitsi-meet as well see:  [jitsi/lib-jitsi-meet#2305](https://github.com/jitsi/lib-jitsi-meet/pull/2305) and #13508
> 
> Agreed. It should also be less code I think.
> 
> @subhamcyara can you pl move it to LJM, following the lead of rtcstats?

@saghul 
As far as I see it, it's okay for lib-jitsi-meet users to not have this functionality. This integration is only regarding the iframe API including JaaS. If any user happens to use lib-jitsi-meet, watchRTC would not require any kind of integration with lib-jitsi-meet as it is required with jitsi-meet to work.

**Comment by saghul** [comment]
While that is technically correct we've seen lib-Jitsi-meet customers ask for a builtin stats system since they want to leverage it.

Since we went in that direction it makes sense that all stats implementations follow suit.

**Comment by subhamcyara** [comment]
Yes totally agree. I would follow the lead of how it is done with rtcstats and modify.
Thanks.

**Comment by subhamcyara** [comment]
@saghul @andrei-gavrilescu 
As directed, I have modified to move watchrtc to LJM. Also on LJM's side https://github.com/jitsi/lib-jitsi-meet/pull/2311
Please let me know your feedback.
Thanks.

**Comment by saghul** [comment]
Thank you!

**Comment by andrei-gavrilescu** [comment]
LGTM!

**Review by saghul — APPROVED** [review]
LGTM

**Comment by damencho** [comment]
@saghul Can we remove the error log we keep printing after this PR:
```
2023-07-20T21:41:40.910Z [modules/watchRTC/WatchRTC.ts] <Object.start>:  WatchRTC is enabled but it has not been configured.
```
I was commenting about it earlier but seems this was not fixed.

**Comment by damencho** [comment]
Actually the print is in ljm PR

**Comment by subhamcyara** [comment]
https://github.com/jitsi/lib-jitsi-meet/pull/2325
I took care of the log here. Apologies for missing it.

**Pull request merged**